### DISCUSSION TITLE
How to run serenity BDD as single executable jar?

### DISCUSSION THREAD

**Discussion by guptadeepak0687** [serenity-bdd/serenity-core#2866] [General]
Is there any way to execute serenity BDD test cases as single executable jar?
I want to use this feature to run in offline mode from ansible.

**Comment by globalworming** [comment]
there may be a way to create a fat jar and execute it https://stackoverflow.com/a/36047816

**Reply by guptadeepak0687** [reply]
Thanks for your reference.
I already tried this options this is for JUnit, but in serenity BDD  there is no main method to get into, have serenity runner instead, so how can we use that as main, thats where i  stuck

**Reply by wakaleo** [reply]
This isn't really supported so you will need to come up with a custom approach like the one in the link above. At the end of the day Serenity tests are just JUnit tests, so a similar approach should work.

**Comment by guptadeepak0687** [comment]
Did you get a chance to look?
